drivers/modem/hl7800.c

```diff
@@ -0,0 +1,4887 @@
+		len = MDM_HL7800_REVISION_MAX_STRLEN;
+	}
+
+	out_len = net_buf_linearize(
+		ictx.mdm_revision, sizeof(ictx.mdm_revision) - 1, *buf, 0, len);
+	ictx.mdm_revision[out_len] = 0;
+	LOG_INF("Revision: %s", log_strdup(ictx.mdm_revision));
+	event_handler(HL7800_EVENT_REVISION, ictx.mdm_revision);
+done:
+	return true;
+}
+
+/* Handler: AT+CGSN */
+static bool on_cmd_atcmdinfo_imei(struct net_buf **buf, uint16_t len)
+{
+	struct net_buf *frag = NULL;
+	size_t out_len;
+
+	/* make sure IMEI data is received
+	 *  waiting for: ###############\r\n
+	 */
+	wait_for_modem_data_and_newline(buf, net_buf_frags_len(*buf),
+					MDM_HL7800_IMEI_SIZE);
+
+	frag = NULL;
+	len = net_buf_findcrlf(*buf, &frag);
+	if (!frag) {
+		LOG_ERR("Unable to find IMEI end");
+		goto done;
+	}
+	if (len < MDM_HL7800_IMEI_STRLEN) {
+		LOG_WRN("IMEI too short (len:%d)", len);
+	} else if (len > MDM_HL7800_IMEI_STRLEN) {
+		LOG_WRN("IMEI too long (len:%d)", len);
+		len = MDM_HL7800_IMEI_STRLEN;
+	}
+
+	out_len = net_buf_linearize(ictx.mdm_imei, sizeof(ictx.mdm_imei) - 1,
+				    *buf, 0, len);
+	ictx.mdm_imei[out_len] = 0;
+
+	LOG_INF("IMEI: %s", log_strdup(ictx.mdm_imei));
+done:
+	return true;
+}
+
+/* Handler: +CCID: <ICCID> */
+static bool on_cmd_atcmdinfo_iccid(struct net_buf **buf, uint16_t len)
+{
+	struct net_buf *frag = NULL;
+	size_t out_len;
+
+	/* make sure ICCID data is received
+	 *  waiting for: <ICCID>\r\n
+	 */
+	wait_for_modem_data_and_newline(buf, net_buf_frags_len(*buf),
+					MDM_HL7800_ICCID_SIZE);
+
+	frag = NULL;
+	len = net_buf_findcrlf(*buf, &frag);
+	if (!frag) {
+		LOG_ERR("Unable to find ICCID end");
+		goto done;
+	}
+	if (len > MDM_HL7800_ICCID_STRLEN) {
+		LOG_WRN("ICCID too long (len:%d)", len);
+		len = MDM_HL7800_ICCID_STRLEN;
+	}
+
+	out_len = net_buf_linearize(ictx.mdm_iccid, MDM_HL7800_ICCID_STRLEN,
+				    *buf, 0, len);
+	ictx.mdm_iccid[out_len] = 0;
+
+	LOG_INF("ICCID: %s", log_strdup(ictx.mdm_iccid));
+done:
+	return true;
+}
+
+static void dns_work_cb(struct k_work *work)
+{
+#ifdef CONFIG_DNS_RESOLVER
+	int ret;
+
+	/* set new DNS addr in DNS resolver */
+	LOG_DBG("Refresh DNS resolver");
+	struct dns_resolve_context *dnsCtx = dns_resolve_get_default();
+	dns_resolve_close(dnsCtx);
+	const char *dns_servers_str[] = { ictx.dns_string };
+	ret = dns_resolve_init(dnsCtx, dns_servers_str, NULL);
+	if (ret < 0) {
+		LOG_ERR("dns_resolve_init fail (%d)", ret);
+		return;
+	}
+#endif
+}
+
+char *mdm_hl7800_get_iccid(void)
+{
+	return ictx.mdm_iccid;
+}
+
+char *mdm_hl7800_get_sn(void)
+{
+	return ictx.mdm_sn;
+}
+
+char *mdm_hl7800_get_imei(void)
+{
+	return ictx.mdm_imei;
+}
+
+char *mdm_hl7800_get_fw_version(void)
+{
+	return ictx.mdm_revision;
+}
+
+/* Handler: +CGCONTRDP: <cid>,<bearer_id>,<apn>,<local_addr and subnet_mask>,
+ *			<gw_addr>,<DNS_prim_addr>,<DNS_sec_addr>
+ */
+static bool on_cmd_atcmdinfo_ipaddr(struct net_buf **buf, uint16_t len)
+{
+	int ret;
+	int numDelims = 7;
+	char *delims[numDelims];
```

Comment:
In order to get rid of this VLA, could you have here something like this instead:
```
#define MAX_DELIMS 7
int numDelims = MAX_DELIMS;
char *delims[MAX_DELIMS];
```
The compiler might be able to set this correctly already but just in case better to be in the safe side.